app/routes/circulars._archive._index/SynonymGroupIndex.tsx

```diff
@@ -5,17 +5,70 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-import { Link } from '@remix-run/react'
+import { Link, useFetcher } from '@remix-run/react'
+import type { action } from 'app/routes/synonyms/route'
+import { useEffect, useState } from 'react'
 
-import type { SynonymGroupWithMembers } from '../synonyms/synonyms.lib'
+import type { Circular } from '../circulars/circulars.lib'
+import type { SynonymGroup } from '../synonyms/synonyms.lib'
+import Spinner from '~/components/Spinner'
+
+function SynonymList({
+  eventIds,
+  slugs,
+  searchString,
+}: {
+  eventIds: string[]
+  slugs: string[]
+  searchString: string
+}) {
+  const [items, setItems] = useState<Circular[]>([])
+  const fetcher = useFetcher<typeof action>()
+  const data = new FormData()
+  data.set('eventIds', eventIds.join(','))
+
+  useEffect(() => {
+    setItems(fetcher.data ?? [])
+  }, [fetcher.data])
+
+  return (
+    <div>
+      <details
+        onClick={() => {
+          fetcher.submit(data, { method: 'POST', action: '/synonyms' })
```

Comment:
Please use `GET` rather than `POST`, since this request does not mutate any data.

That might mean that you have to hang the loader off of a different or a new route.

Reply:
I made it a post because we are posting data. You can't post data with a GET. It would then have to be in a query string, which doesn't support arrays. GETs don't have a body. So the options are to send it as a query string or to use POST and post the array of eventIds.

Reply:
> I made it a post because we are posting data.

Yes, but _why_ are you posting the data?

> It would then have to be in a query string, which doesn't support arrays.

Both POST with URL form encoding and GET with query strings support multiple form key-value pairs with the same keys but different values. (After all, they are the same encoding!)

```
> let params = new URLSearchParams()
undefined
> params.append('foo', 'bar')
undefined
> params.append('foo', 'bat')
undefined
> params.toString()
'foo=bar&foo=bat'
```